hnn_core/feed.py

```diff
@@ -35,252 +74,238 @@ class ExtFeed(object):
         Parameters of the external input feed, arranged into a dictionary.
     gid : int
         The cell ID.
+    """
+    prng, prng2 = _get_prng(
+        seed=params['prng_seedcore'],
+        gid=gid,
+        sync_evinput=params.get('sync_evinput', False))
 
-    Attributes
+    # check feed name validity, allowing substring matches ('evprox1' etc)
+    valid_feeds = ['extpois', 'extgauss', 'common', 'evprox', 'evdist']
+    # NB check if feed_type has a valid substring, not vice versa
+    matches = [f for f in valid_feeds if f in feed_type]
+    if len(matches) == 0:
+        raise ValueError('Invalid external feed: %s' % feed_type)
+    elif len(matches) > 1:
+        raise ValueError('Ambiguous external feed: %s' % feed_type)
+
+    # Return values not checked: False if all weights for given feed type
+    # are zero. Designed to be silent so that zeroing input weights
+    # effectively disables each.
+    zero_ampa_nmda = False
```

Comment:
This is nitpicky, but could we rename `zero_ampa_nmda` to `target_syn_weights_zero` just to make it consistent with `all_sysn_weights_zero`? This way it's super clear which one applies to unique feeds and which one applies to common feeds.